test/test_torch.py

```diff
@@ -2519,6 +2526,57 @@ def test_serialization(self):
             self.assertEqual(c[1], c[3], 0)
             self.assertEqual(c[4], c[5][1:4], 0)
 
+            # check that serializing the same storage view object unpickles
+            # it as one object not two (and vice versa)
+            views = c[7]
+            self.assertEqual(views[0]._cdata, views[1]._cdata)
+            self.assertEqual(views[0], views[2])
+            self.assertNotEqual(views[0]._cdata, views[2]._cdata)
```

Comment:
No; I'm precisely trying to check that the cdata is the same, not that the python objects are the same. In fact, the python object should not be the same (every call to `t.storage()` produces a different python object). Python object deduplication is guaranteed by pickle; here I'm testing C object deduplication, which I have to do explicitly in the save/load code.